src/robotide/controller/commands.py

```diff
@@ -279,12 +280,35 @@ class NullObserver(object):
 class RenameKeywordOccurrences(_ReversibleCommand):
 
     def __init__(self, original_name, new_name, observer, keyword_info=None):
-        self._original_name = original_name
-        self._new_name = new_name
+        keyword_name, new_keyword_name = self._check_gherkin(new_name,
+                                                             original_name)
+        self._original_name = keyword_name
+        self._new_name = new_keyword_name
         self._observer = observer
         self._keyword_info = keyword_info
         self._occurrences = None
 
+    def _check_gherkin(self, new_name, original_name):
+        _gherkin_prefix = re.compile('^(Given|When|Then|And|But) ',
```

Comment:
On a related note, I think the regex should be a class variable, since then it would only be compiled once.